### PR TITLE
Docs and news for #38952

### PR DESCRIPTION
**Pull request by DilumAluthge**
This adds docs and news for #38952 

Also, the docs had a statement that is now outdated by https://github.com/JuliaLang/julia/pull/36778, so I have removed it.

### CODE REVIEW

File: doc/src/manual/environment-variables.md [hunk 1]
```diff
@@ -190,19 +190,23 @@ a master process to establish a connection before dying.
 ### [`JULIA_NUM_THREADS`](@id JULIA_NUM_THREADS)
 
 An unsigned 64-bit integer (`uint64_t`) that sets the maximum number of threads
-available to Julia. If `$JULIA_NUM_THREADS` exceeds the number of available
-CPU threads (logical cores), then the number of threads is set to the number of CPU threads. If
-`$JULIA_NUM_THREADS` is not positive or is not set, or if the number of CPU
-threads cannot be determined through system calls, then the number of threads is
-set to `1`.
+available to Julia.  If `$JULIA_NUM_THREADS` is not positive or is not set, or
```
Comment (JeffBezanson) [inline doc/src/manual/environment-variables.md:193]: The change to this paragraph should be backported to 1.6.

**Comment by JeffBezanson** [comment]
Note whitespace error in CI.

File: doc/src/manual/environment-variables.md (hunk 1 above)
Comment (DilumAluthge) [reply doc/src/manual/environment-variables.md:193]: Sounds good. I'll make a PR to `release-1.6` to fix just the part of the paragraph that applies to 1.6.

**Comment by DilumAluthge** [comment]
> Note whitespace error in CI.

Alright, whitespace should be fixed now,

File: doc/src/manual/environment-variables.md (hunk 1 above)
Comment (DilumAluthge) [reply doc/src/manual/environment-variables.md:193]: Here's the PR for 1.6: https://github.com/JuliaLang/julia/pull/39118

**Comment by DilumAluthge** [comment]
Is this good to merge now?

**Pull request merged**